scoring_engine/engine/engine.py

```diff
@@ -251,7 +253,12 @@ def run(self):
 
             if not self.is_last_round():
                 round_time_sleep = int(Setting.get_setting('round_time_sleep').value)
-                logger.info("Sleeping in between rounds (" + str(round_time_sleep) + " seconds)")
-                self.sleep(round_time_sleep)
+                round_duration = (datetime.now() - round_start_time).seconds
+                round_delta = round_time_sleep - round_duration
+                if round_delta > 0:
+                    logger.info("Sleeping in between rounds (" + str(round_delta) + " seconds)")
+                    self.sleep(round_delta)
+                else:
+                    logger.warning("Service checks lasted " + str(-round_delta) + "s longer than round length. Starting next round immediately")
```

Comment:
Should probably remove the "-", otherwise it looks funny...

Service checks lasted -44s longer than round length. Starting next round immediately

Reply:
the else case means round_delta is negative, so this just brings it back into the positive realm for printing

Reply:
ah. for my own sanity, can you use abs() instead?